includes/class-wc-payments-features.php

```diff
@@ -210,7 +210,15 @@ public static function is_woopay_eligible() {
 
 		// read directly from cache, ignore cache expiration check.
 		$account = WC_Payments::get_database_cache()->get( WCPay\Database_Cache::ACCOUNT_KEY, true );
-		return is_array( $account ) && ( $account['platform_checkout_eligible'] ?? false );
+
+		$is_account_rejected = WC_Payments::get_account_service()->is_account_rejected();
+
+		$is_account_under_review = WC_Payments::get_account_service()->is_account_under_review();
```

Comment:
I created this issue only taking into account "rejected*" and "under_review" statuses, but it looks like there could be [more statuses](https://github.com/Automattic/woocommerce-payments-server/blob/9e3b5c29913af25da82e7efb7289e15d72a83ef8/server/wp-content/rest-api-plugins/endpoints/wcpay/stripe/class-stripe-account.php#L26-L31) and I'm not sure if/how these would be set in a real scenario.

@asumaran should we consider those other statuses as well? Perhaps untimately it would be easier to just check if the account is enabled/complete/not currently restricted?

Reply:
> should we consider those other statuses as well? 

Good question. We'd need to confirm what statuses actually block the account. I'll double check tomorrow. In the meantime I'll ask internally about that.

Reply:
I noticed the account status is set to `restricted` in two cases:

- When payments are paused on Stripe.
- When payouts are paused on Stripe.

When the merchant's account is `restricted` WooPayments is not disabled on the site neither the WooPayments account.

Does it make sense to turn off the merchant's WooPay eligibility if their Stripe account is restricted? cc @pierorocca @bborman22

Reply:
The code on WooPayments is actually working as expected. I've removed most of the code I was going to introduce. 

The WooPay webhook is not registered if the account is suspended or rejected. Here in my PR I'm just removing the webhook in case the account is not eligible.

The question above still stands but it should be implemented in the [server](https://github.com/Automattic/woocommerce-payments-server/pull/5453) instead.

Reply:
I've tested with a JN site + my sandbox and can confirm WooPay doesn't work if the merchant has payments disabled on Stripe. No iframe is triggered after entering a valid WooPay email. The account status is set to `restricted`.

The server will set the site as not eligible for WooPay. So, nothing to add there as it's already being handled correctly.

>Perhaps untimately it would be easier to just check if the account is enabled/complete/not currently restricted?

I think disabling WooPay for "rejected", "under review" and "restricted" accounts it's OK from WooPayments. If there's more status we need to handle it would be better to do it server side.